GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -584,8 +624,19 @@ private void PostFetchJob(List<string> packIndexes)
 
                     using (ITracer activity = this.context.Tracer.StartActivity("TryWriteGitCommitGraph", EventLevel.Informational, Keywords.Telemetry, metadata: null))
                     {
-                        GitProcess process = new GitProcess(this.context.Enlistment);
-                        GitProcess.Result result = process.WriteCommitGraph(this.context.Enlistment.GitObjectsRoot, packIndexes);
+                        this.backgroundGitProcess = new GitProcess(this.context.Enlistment);
+
+                        if (this.stopping)
+                        {
+                            this.context.Tracer.RelatedWarning(
+                                metadata: null,
+                                message: PostFetchTelemetryKey + ": Not launching 'git commit-graph' because the mount is stopping",
+                                keywords: Keywords.Telemetry);
+                            return;
+                        }
+
+                        GitProcess.Result result = this.backgroundGitProcess.WriteCommitGraph(this.context.Enlistment.GitObjectsRoot, packIndexes);
+                        this.backgroundGitProcess = null;
 
                         if (result.HasErrors)
                         {

```

Comment:
I can't put this comment on the correct line but - do we still need the `ThreadAbortException` catch block below?